src/crypto/ed448_suite.c

```diff
@@ -0,0 +1,205 @@
+{
+    if (!signature || !ec_public_key || !data_to_verify)
+        return false;
+
+    if (signature->curve.identifier != EC_CURVE_ID_ED448_GOLDILOCKS ||
+        ec_public_key->curve.identifier != EC_CURVE_ID_ED448_GOLDILOCKS)
+        return false;
+
+    const wickr_buffer_t * raw_signature = signature->sig_data;
+    if (!raw_signature || raw_signature->length != EDDSA_448_SIGNATURE_LENGTH)
+        return false;
+    
+    wickr_buffer_t *public_data = ec_public_key->pub_data;
+
+    if (!public_data || public_data->length != EDDSA_448_PUBLIC_KEY_LENGTH)
+        return false;
+
+    uint8_t prehashed = 0;
+    uint8_t *context = NULL;
+    uint8_t context_length = 0;
+
+    return (decaf_ed448_verify(raw_signature->bytes, public_data->bytes, data_to_verify->bytes,
+            data_to_verify->length, prehashed, context, context_length) == DECAF_SUCCESS);
+
+}
+
+wickr_buffer_t *ed448_dh_derive_public_key(const wickr_buffer_t *private_key_data)
+{
+    if (!private_key_data || private_key_data->length != DH_448_PRIVATE_KEY_LENGTH)
+        return NULL;
+
+    wickr_buffer_t *public_key_data = wickr_buffer_create_empty_zero(DH_448_PUBLIC_KEY_LENGTH);
+
+    if (!public_key_data)
+        return NULL;
+
+    decaf_x448_derive_public_key(public_key_data->bytes, private_key_data->bytes);
+    // This function does not allow for failure
+
+    return public_key_data;  
+}
+
+wickr_buffer_t *ed448_dh_shared_secret(const wickr_ecdh_params_t *params)
+{
+    if (!params)
+        return NULL;
+    
+    const wickr_ec_key_t *local_key_pair = params->local_key;
+    const wickr_ec_key_t *peer_public_key = params->peer_key;
+
+    if (!local_key_pair || !peer_public_key)
+        return NULL;
+
+    if (local_key_pair->curve.identifier != EC_CURVE_ID_ED448_GOLDILOCKS ||
+        peer_public_key->curve.identifier != EC_CURVE_ID_ED448_GOLDILOCKS)
+        return NULL;
+        
+    const wickr_buffer_t *local_private_key_data = local_key_pair->pri_data;
+    const wickr_buffer_t *peer_public_key_data = peer_public_key->pub_data;
+
+    if (!local_private_key_data || !peer_public_key_data)
+        return NULL;
+
+    if (local_private_key_data->length != DH_448_PRIVATE_KEY_LENGTH ||
+        peer_public_key_data->length != DH_448_PUBLIC_KEY_LENGTH)
+        return NULL;
+
+    wickr_buffer_t *shared_secret = wickr_buffer_create_empty_zero(DH_448_SHARED_SECRET_LENGTH);
+
+    /* Compute the shared secret */
+    decaf_error_t result = decaf_x448(shared_secret->bytes, peer_public_key_data->bytes,
+                                      local_private_key_data->bytes);
+    
+    if (result == DECAF_FAILURE) {
+        wickr_buffer_destroy(&shared_secret);
+        return NULL;
+    }
+
+    /* Run the ECDH shared secret through HKDF with the provided salt and info from the ECDH params */
+    wickr_kdf_result_t *kdf_result = wickr_perform_kdf_meta(params->kdf_info, shared_secret);
+    wickr_buffer_t *final_buffer = wickr_buffer_copy(kdf_result->hash);
```

Comment:
shared secret should be destroyed here to a avoid a memory leak, it won't be destroyed by the wickr_perform_kdf_meta function

Reply:
Right!